cirq-core/cirq/ops/matrix_gates.py

```diff
@@ -114,11 +114,11 @@ def with_name(self, name: str) -> 'MatrixGate':
         return MatrixGate(self._matrix, name=name, qid_shape=self._qid_shape, unitary_check=False)
 
     def _json_dict_(self) -> Dict[str, Any]:
-        return {'matrix': self._matrix.tolist(), 'qid_shape': self._qid_shape}
+        return {'matrix': self._matrix.tolist(), 'qid_shape': self._qid_shape, **({'name': self._name} if self._name else {})}
```

Comment:
Strictly speaking this sends empty string to `None` after serialization+deserialization.

Reply:
Do we consider an empty string as a valid name? If so, I’ll update the check to more explicitly check None.

Reply:
Well, at present the empty string is accepted by the ctor, so let's make sure it works in serde, too.

Reply:
Good point. I've updated the check logic and added an empty name test.